lib/srv/db/proxyserver.go

```diff
@@ -625,9 +626,16 @@ func (s *ProxyServer) getConfigForServer(ctx context.Context, identity tlsca.Ide
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	teleportVer, err := semver.NewVersion(server.GetTeleportVersion())
+	if err != nil {
+		return nil, trace.WrapWithMessage(err, "failed to parse Teleport version")
+	}
+
 	response, err := s.cfg.AuthClient.SignDatabaseCSR(ctx, &proto.DatabaseCSRRequest{
-		CSR:         csr,
-		ClusterName: identity.RouteToCluster,
+		CSR:                csr,
+		ClusterName:        identity.RouteToCluster,
+		SignWithDatabaseCA: teleportVer.LessThan(*semver.New("9.1.0")),
```

Comment:
We need to check here an agent teleport version that we are going to connect not teleport proxy version.